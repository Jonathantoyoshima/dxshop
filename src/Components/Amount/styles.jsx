import styled from 'styled-components';

export const Root = styled.div`
  display: flex;
`;

export const Input = styled.input`
  background: transparent;
  border: 0;
  width: 25px;
  text-align: center;
  color: ${(props) => props.theme.color.primary.light};
`;

export const Button = styled.button`
  background-color: ${(props) => props.theme.color.primary.light};
  border: 0;
  border-radius: 10px;
  color: ${(props) => props.theme.color.primary.main};
  cursor: pointer;
  height: 20px;
  width: 20px;

  :hover {
    background-color: ${(props) => props.theme.color.primary.main};
    color: ${(props) => props.theme.color.primary.light};
  }
`;
