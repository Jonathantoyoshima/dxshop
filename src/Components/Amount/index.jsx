import { Root, Button, Input } from './styles';

const Amount = () => {
  return (
    <Root>
      <Button>-</Button>
      <Input value={1} onChange={() => {}} />
      <Button>+</Button>
    </Root>
  );
};
const props = {
  theme: {
    color: {
      basic: {
        black: '#0c0c0c',
        white: '#fff',
      },
      primary: {
        light: '#000',
        main: '#ccc',
        dark: '#000',
      },
    },
  },
};
Input.defaultProps = props;
Button.defaultProps = props;

export default Amount;
