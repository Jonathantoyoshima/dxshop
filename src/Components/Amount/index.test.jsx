import React from 'react';
import renderer from 'react-test-renderer';
import Amount from './index';

test('Render Build Amount Snapshot', () => {
	const component = renderer.create(<Amount />);
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
