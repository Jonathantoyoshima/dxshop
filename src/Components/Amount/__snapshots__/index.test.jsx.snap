// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Build Amount Snapshot 1`] = `
<div
  className="sc-bdfBwQ gbvaeM"
>
  <button
    className="sc-dlfnbm ieqtFq"
  >
    -
  </button>
  <input
    className="sc-gsTCUz fhZvdj"
    onChange={[Function]}
    value={1}
  />
  <button
    className="sc-dlfnbm ieqtFq"
  >
    +
  </button>
</div>
`;
